boardswarm-cli/src/main.rs

```diff
@@ -242,6 +242,76 @@ enum AuthCommand {
     Modify(AuthModifyArg),
 }
 
+#[derive(Clone, Debug)]
+enum ItemArg {
+    Id(u64),
+    Name(String),
+}
+
+async fn item_lookup(
+    arg: ItemArg,
+    item_type: ItemType,
+    mut client: Boardswarm,
+) -> Result<u64, anyhow::Error> {
+    let item_kind = match item_type {
+        ItemType::Device => ("Device", "device"),
+        ItemType::Console => ("Console", "console"),
+        ItemType::Actuator => ("Actuator", "actuator"),
+        ItemType::Volume => ("Volume", "volume"),
+    };
+    match arg {
+        ItemArg::Id(id) => Ok(id),
+        ItemArg::Name(name) => {
+            let mut items = client.list(item_type).await?;
+
+            let name_components: Vec<_> = name.split('@').collect();
+            if name_components.len() > 2 {
+                return Err(anyhow!("Too many @ in {} name", item_kind.1));
+            }
+
+            /*
+             * first try finding a single exact match for the "name"
+             * part of the user-supplied "name[@instance]"
+             */
+            items.retain(|i| i.name == name_components[0]);
```

Comment:
this would retrain items regardless of whether the instance matches... And you don't end up checking for that later :)

If you want the semantic of the instance being optional you can use e.g. `i .name== name && (instance.is_none()  || i.instance == instance)` otherwise just `i.name == name && i.instance == instance`